modules/EvseV2G/iso_server.cpp

```diff
@@ -924,11 +927,11 @@ static enum v2g_event handle_iso_service_discovery(struct v2g_connection* conn)
     /* Find requested scope id within evse service list */
     if (req->ServiceScope_isUsed) {
         /* Check if ServiceScope is in evse ServiceList */
-        for (uint8_t idx = 0; idx < res->ServiceList.Service.arrayLen; idx++) {
-            if ((res->ServiceList.Service.array[idx].ServiceScope_isUsed == (unsigned int)1) &&
-                (strcmp(res->ServiceList.Service.array[idx].ServiceScope.characters, req->ServiceScope.characters) ==
-                 0)) {
-                scope_idx = idx;
+        for (uint16_t idx = 0; idx < evse_service_list_len; idx++) {
+            if ((conn->ctx->evse_service_list.array[idx].ServiceScope_isUsed == (unsigned int)1) &&
+                (strncmp(conn->ctx->evse_service_list.array[idx].ServiceScope.characters, req->ServiceScope.characters,
+                         sizeof(req->ServiceScope.characters)) == 0)) {
```

Comment:
Please use strcmp and change to `conn->ctx->evse_v2g_data.evse_service_list[idx].ServiceScope.characters`